src/lib/scan/scan_queue.js

```diff
@@ -81,6 +107,43 @@ export class ScanQueue {
     this._isManualScan = false;
   }
 
+  /**
+   * Updates scan state and notifies listeners.
+   *
+   * @param {boolean} isScanning - True if scanning is active.
+   * @param {number} scannedCount - Number of already scanned items.
+   * @private
+   */
+  _changeScanState(isScanning, scannedCount) {
+    this._isScanning = isScanning;
+    this._scanCompleteCount = scannedCount;
+    if (this._queueStateChangeHandler != null) {
+      this._queueStateChangeHandler(this.getScanState());
+    }
+  }
+
+  /**
+   * Returns current scan state.
+   *
+   * @returns {{state: string, queueLength: number, scanned: number}} Scan
+   *   queue state enum.
+   */
+  getScanState() {
+    if (this._isScanning) {
```

Comment:
Could replace this with a single return statement and `state: this_isScanning ? scanQueueStateEnum.ACTIVE : scanQueueStateEnum.INACTIVE,`